### PR TITLE
Climbing down onto the tables from another z-level.

### PR DESCRIPTION
**Pull request by Tea-Primoris**
## About The Pull Request
You can descend onto tables.
mOed0I7Da6

## Why It's Good For The Game
You can climb up, and now you can climb down without breaking bones.

## Changelog
:cl:
add: Now you can climb down onto tables from z-level above by dragging yourself onto openspace.
/:cl:

### CODE REVIEW

**Comment by Hopekz** [comment]
Good change

File: code/modules/multiz/turf.dm [hunk 1]
```diff
@@ -240,3 +240,49 @@ see multiz/movement.dm for some info.
...
+		to_chat(user, SPAN_DANGER("There's \a [occupied] in the way."))
+		return
+
+	return 1
+
+/turf/simulated/open/proc/do_descent(var/mob/living/user, var/obj/structure/structure)
+	if(!can_descent(user, structure))
+		return
+
+	usr.visible_message(SPAN_WARNING("[user] starts descenting onto [structure]!"))
```
Comment (mattroks101) [inline code/modules/multiz/turf.dm:270]:
```suggestion
	user.visible_message(SPAN_WARNING("[user] starts descending onto [structure]!"))
```

File: code/modules/multiz/turf.dm [hunk 2]
```diff
@@ -240,3 +240,49 @@ see multiz/movement.dm for some info.
...
+	var/duration = max(delay * user.stats.getMult(STAT_VIG, STAT_LEVEL_EXPERT), delay * 0.66)
+	if(!do_after(user, duration, src))
+		climbers -= user
+		return
+
+	if(!can_descent(user, structure, post_descent_check = 1))
+		climbers -= user
+		return
+
+	usr.forceMove(GetBelow(src))
```
Comment (mattroks101) [inline code/modules/multiz/turf.dm:284]:
```suggestion
	user.forceMove(GetBelow(src))
```

File: code/modules/multiz/turf.dm [hunk 3]
```diff
@@ -240,3 +240,49 @@ see multiz/movement.dm for some info.
...
+
+	if(get_turf(user) == GetBelow(src))
+		usr.visible_message(SPAN_WARNING("[user] descents onto [structure]!"))
```
Comment (mattroks101) [inline code/modules/multiz/turf.dm:287]:
```suggestion
		user.visible_message(SPAN_WARNING("[user] descends onto [structure]!"))
```

File: code/modules/multiz/turf.dm [hunk 4]
```diff
@@ -240,3 +240,49 @@ see multiz/movement.dm for some info.
...
+		do_descent(target, S)
+	else
+		return ..()
+
+/turf/simulated/open/proc/can_descent(var/mob/living/user, var/obj/structure/structure, post_descent_check = 0)
+	if(!structure || !structure.climbable || (!post_descent_check && (user in climbers)))
+		return
+
+	if(!user.Adjacent(src))
+		to_chat(user, SPAN_DANGER("You can't descent there, the way is blocked."))
```
Comment (mattroks101) [inline code/modules/multiz/turf.dm:256]:
```suggestion
		to_chat(user, SPAN_DANGER("You can't descend there, the way is blocked."))
```

File: code/modules/multiz/turf.dm [hunk 5]
```diff
@@ -240,3 +240,49 @@ see multiz/movement.dm for some info.
...
+		return
+
+	return 1
+
+/turf/simulated/open/proc/do_descent(var/mob/living/user, var/obj/structure/structure)
+	if(!can_descent(user, structure))
+		return
+
+	usr.visible_message(SPAN_WARNING("[user] starts descenting onto [structure]!"))
+	structure.visible_message(SPAN_WARNING("Someone starts descenting onto [structure]!"))
```
Comment (mattroks101) [inline code/modules/multiz/turf.dm:271]:
```suggestion
	structure.visible_message(SPAN_WARNING("Someone starts descending onto [structure]!"))
```

**Pull request merged**